Dockerfile

```diff
@@ -0,0 +1,35 @@
+# Autoretrieve uses Docker Buildkit to speed up builds, make sure
+# DOCKER_BUILDKIT=1 is set
+
+FROM golang:1.17
+
+RUN --mount=type=cache,target=/var/cache/apt \
+    apt update && \
+    apt install -y --no-install-recommends jq libhwloc-dev ocl-icd-opencl-dev
+
+# Make sure extern deps are ready first
+WORKDIR /app
+COPY .git .git
+COPY .gitmodules .gitmodules
```

Comment:
Won't copying the `.git` directory counteract the caching, since every commit will change this directory, and docker will need rebuild a lower layer each time?

Reply:
Was just in the middle of writing a comment about that, hah. Moving .git into the image isn't great, anyway, and I'd be more than happy to leave it out.

I thought about just ignoring the .git directory and then cloning filecoin-ffi from scratch into extern/filecoin-ffi like would be done with submodules, but that gets rid of the version locking. I'd love to move away from using submodules for extern dependencies altogether.

I still don't know a better alternative for extern deps. Maybe a makefile that can download the correct version? Feedback appreciated.

Reply:
Was just in the middle of writing a comment about that, hah. Moving .git into the image isn't great, anyway, and I'd be more than happy to leave it out.

I thought about just ignoring the .git directory and then cloning filecoin-ffi from scratch into extern/filecoin-ffi like would be done with submodules, but that gets rid of the version locking. I'd love to move away from using submodules for extern dependencies altogether.

I still don't know a better alternative for extern deps. Maybe a makefile that can download the correct version? Feedback appreciated.